autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/ui/credential/management/AutofillManagementRecyclerAdapter.kt

```diff
@@ -60,51 +68,82 @@ class AutofillManagementRecyclerAdapter(
+        val item = listItems[position] as SuggestedCredential
+        populateCredentialsDetails(item.credentials, viewHolder)
+    }
+
+    private fun populateCredentialsDetails(loginCredentials: LoginCredentials, viewHolder: CredentialsViewHolder) {
         with(viewHolder.binding) {
-            title.text = titleExtractor.extract(item.credentials)
-            subtitle.text = item.credentials.username
-            root.setOnClickListener { onCredentialSelected(item.credentials) }
+            title.text = titleExtractor.extract(loginCredentials)
+            subtitle.text = loginCredentials.username
+            root.setOnClickListener { onCredentialSelected(loginCredentials) }
 
-            val popupMenu = initializePopupMenu(root.context, item.credentials)
+            val popupMenu = initializePopupMenu(root.context, loginCredentials)
             overflowMenu.setOnClickListener {
                 popupMenu.show(root, it)
             }
 
-            updateFavicon(item.credentials)
+            updateFavicon(loginCredentials)
         }
     }
 
     private fun onBindViewHolderHeading(position: Int, viewHolder: HeadingViewHolder) {
-        val item = listItems[position] as ListItem.GroupHeading
+        val item = listItems[position] as GroupHeading
         with(viewHolder.binding) {
-            groupHeader.text = item.initial.toString()
+            groupHeader.primaryText = item.initial
         }
     }
 
     override fun getItemViewType(position: Int): Int {
         return when (listItems[position]) {
-            is ListItem.GroupHeading -> ITEM_VIEW_TYPE_HEADER
-            is ListItem.Credential -> ITEM_VIEW_TYPE_CREDENTIAL
+            is GroupHeading -> ITEM_VIEW_TYPE_HEADER
+            is Credential -> ITEM_VIEW_TYPE_CREDENTIAL
+            is SuggestedCredential -> ITEM_VIEW_TYPE_SUGGESTED_CREDENTIAL
+            is Divider -> ITEM_VIEW_TYPE_DIVIDER
         }
     }
 
     private fun initializePopupMenu(context: Context, loginCredentials: LoginCredentials): PopupMenu {
-        return PopupMenu(LayoutInflater.from(context), R.layout.overflow_menu_list_item).apply {
+        return PopupMenu(LayoutInflater.from(context), R.layout.overflow_menu_list_item, width = getPopupMenuWidth(context)).apply {
             onMenuItemClicked(contentView.findViewById(R.id.item_overflow_edit)) { onContextMenuItemClicked(Edit(loginCredentials)) }
             onMenuItemClicked(contentView.findViewById(R.id.item_overflow_delete)) { onContextMenuItemClicked(Delete(loginCredentials)) }
+            onMenuItemClicked(contentView.findViewById(R.id.item_copy_username)) { onContextMenuItemClicked(CopyUsername(loginCredentials)) }
+            onMenuItemClicked(contentView.findViewById(R.id.item_copy_password)) { onContextMenuItemClicked(CopyPassword(loginCredentials)) }
+        }
+    }
+
+    private fun getPopupMenuWidth(context: Context): Int {
```

Comment:
This _shouldn't_ be needed ideally but is a temporary patch until `HorizontalDivider` is changed. @malmstein will make the necessary changes outside of this PR.